api/caching/tests/test_caching.py

```diff
@@ -0,0 +1,242 @@
+
+    @unittest.skipIf(not django_settings.ENABLE_VARNISH, 'Varnish is disabled')
+    def test_compare_python_responses_to_varnish_responses(self):
+        querystrings = dict(
+            nodes=[
+                'comments',
+                'children',
+                'files',
+                'registrations',
+                'contributors',
+                'node_links',
+                'root',
+            ],
+            users=[
+                'nodes',
+            ],
+            registrations=[
+                'comments',
+                'children',
+                'files',
+                'registrations',
+                'contributors',
+                'node_links',
+                'root',
+                'registrations',
+                'registered_by',
+                'registered_from',
+            ]
+        )
+
+        querystring_suffix = 'page[size]=10&format=jsonapi'
+
+        data_dict = dict(nodes=dict(),
+                         users=dict(),
+                         comments=dict(),
+                         registrations=dict(), )
+
+        python_data = copy.deepcopy(data_dict)
+        python_authed_data = copy.deepcopy(data_dict)
+        varnish_data = copy.deepcopy(data_dict)
+        varnish_authed_data = copy.deepcopy(data_dict)
+
+        for key, embed_values in querystrings.items():
+            embed_values.sort()
+            original_embed_values = embed_values
+            while len(embed_values) > 0:
+                generated_qs = '&embed='.join(embed_values)
+                python_url = '{}{}/?embed={}&{}&esi=false'.format(
+                    self.local_python_base_url, key, generated_qs,
+                    querystring_suffix)
+                varnish_url = '{}{}/?embed={}&{}&esi=true'.format(
+                    self.local_varnish_base_url, key, generated_qs,
+                    querystring_suffix)
+                python_resp = requests.get(python_url, timeout=120)
+                python_authed_resp = requests.get(python_url,
+                                                  auth=self.authorization,
+                                                  timeout=120)
+
+                varnish_resp = requests.get(varnish_url, timeout=120)
+                varnish_authed_resp = requests.get(varnish_url,
+                                                   auth=self.authorization,
+                                                   timeout=120)
+
+                try:
+                    python_data[key]['_'.join(
+                        embed_values)] = python_resp.json()
+                    self.validate_keys(python_resp.json(),
+                                       original_embed_values)
+                except Exception as ex:
+                    python_data[key]['_'.join(embed_values)] = dict(
+                        error=ex.message)
+                    print 'Failed to get python_url unauthed: {} - {}\n\n\n'.format(
```

Comment:
@sloria It probably should since the assertions wouldn't actually be running. I wasn't sure what to do in this situation. It would indicate a misconfiguration or a problem with travis rather than an **actual** test failure.